adafruit_oauth2.py

```diff
@@ -52,13 +57,13 @@ class OAuth2:  # pylint: disable=too-many-arguments, too-many-instance-attribute
 
     def __init__(
         self,
-        requests,
-        client_id,
-        client_secret,
-        scopes,
-        access_token=None,
-        refresh_token=None,
-    ):
+        requests: adafruit_requests.Session,
+        client_id: str,
+        client_secret: str,
+        scopes: list,
```

Comment:
So one of the cool things about generics is the ability to specify the types of their contents!  This better helps type checkers as well as IDEs understand what is allowed within applicable types.  You can do so by using brackets.  For this list, it's a list that can contain `str`, so you would typing it as:

```python
        scopes: list[str]
```

Unfortunately, the ability to use builtins as generics was added in Python 3.9, and Blinka currently supports Python 3.7 as the minimum.  Therefore, in order to do this, we'll need to use `typing.List`:

```python
        scopes: List[str]
```

Don't forget to import `List` from the `typing` module!